File: microserv/microservices/appointment-service/internal/models/appointment.go
package models

import "time"

type Appointment struct {
	ID              int       `json:"id"`
	PatientID       int       `json:"patient_id" binding:"required"`
	DoctorID        int       `json:"doctor_id" binding:"required"`
	AppointmentDate time.Time `json:"appointment_date" binding:"required"`
	Status          string    `json:"status"`
	Type            string    `json:"type"`
	Location        string    `json:"location"`
	Duration        int       `json:"duration"`
	Description     string    `json:"description"`
	Notes           string    `json:"notes"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type CreateAppointmentRequest struct {
	PatientID       int       `json:"patient_id" binding:"required"`
	DoctorID        int       `json:"doctor_id" binding:"required"`
	AppointmentDate time.Time `json:"appointment_date" binding:"required"`
	Type            string    `json:"type" binding:"required"`
	Location        string    `json:"location" binding:"required"`
	Duration        int       `json:"duration" binding:"required"`
	Description     string    `json:"description"`
}

type UpdateAppointmentRequest struct {
	AppointmentDate time.Time `json:"appointment_date"`
	Status          string    `json:"status"`
	Notes           string    `json:"notes"`
}

type AppointmentSlot struct {
	ID           int       `json:"id"`
	DoctorID     int       `json:"doctor_id"`
	StartTime    time.Time `json:"start_time"`
	EndTime      time.Time `json:"end_time"`
	IsAvailable  bool      `json:"is_available"`
	BookedByID   *int      `json:"booked_by_id"`
	CreatedAt    time.Time `json:"created_at"`
}

type DoctorAvailability struct {
	DoctorID      int              `json:"doctor_id"`
	AvailableSlots []AppointmentSlot `json:"available_slots"`
}

File: microserv/tampilan microservices/app/Models/Appointment.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Appointment extends Model
{
    protected $fillable = [
        'patient_id', 'doctor_id', 'appointment_date', 'status',
        'description', 'notes', 'type', 'location', 'duration'
    ];

    protected $casts = [
        'appointment_date' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Patient relationship
     */
    public function patient()
    {
        return $this->belongsTo(User::class, 'patient_id');
    }

    /**
     * Doctor relationship
     */
    public function doctor()
    {
        return $this->belongsTo(User::class, 'doctor_id');
    }

    /**
     * Medical records from this appointment
     */
    public function medicalRecords()
    {
        return $this->hasMany(MedicalRecord::class, 'appointment_id');
    }

    /**
     * Payment record
     */
    public function payment()
    {
        return $this->hasOne(Payment::class, 'appointment_id');
    }

    /**
     * Prescriptions from this appointment
     */
    public function prescriptions()
    {
        return $this->hasMany(Prescription::class, 'appointment_id');
    }
}